test/helpers/ValueHelpers.spec.ts

```diff
@@ -0,0 +1,212 @@
+        expect(daml.timestamp('2019-05-22T06:43:42Z')).to.deep.equal({
+            valueType: 'timestamp',
+            timestamp: '2019-05-22T06:43:42Z' // this is correctly typed but will be an error at runtime
+        });
+    });
+    test('dates should return the correct number of days from epoch', () => {
+        expect(daml.date(new Date(0))).to.deep.equal({
+            valueType: 'date',
+            date: '0'
+        });
+    });
+    test('dates should return the correct number of days from the millisecond after epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date(1))).to.deep.equal({
+            valueType: 'date',
+            date: '0'
+        });
+    });
+    test('dates should return the correct number of days from the millisecond before epoch', () => {
+        expect(daml.date(new Date(-1))).to.deep.equal({
+            valueType: 'date',
+            date: '-1'
+        });
+    });
+    test('dates should return the correct number of days from two milliseconds before epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date(-2))).to.deep.equal({
+            valueType: 'date',
+            date: '-1'
+        });
+    });
+    test('dates should return the correct number of microseconds from dates after epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date(86400000))).to.deep.equal({
+            valueType: 'date',
+            date: '1'
+        });
+    });
+    test('dates should return the correct number of microseconds from dates before epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date(-86400000))).to.deep.equal({
+            valueType: 'date',
+            date: '-1'
+        });
+    });
+    test('dates should return the correct number of microseconds from dates before epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date(-86400001))).to.deep.equal({
+            valueType: 'date',
+            date: '-2'
+        });
+    });
+    test('dates should be correctly parsed from dates in ISO 8601 compliant format, after epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date('1970-01-02T06:43:42Z'))).to.deep.equal({
+            valueType: 'date',
+            date: '1'
+        });
+    });
+    test('dates should be correctly parsed from dates in ISO 8601 compliant format, before epoch, losing infra-day precision', () => {
+        expect(daml.date(new Date('1969-12-31T06:43:42Z'))).to.deep.equal({
+            valueType: 'date',
+            date: '-1'
+        });
+    });
+    test('dates should go through untouched when passed as a string', () => {
+        expect(daml.date('2019-05-22T06:43:42Z')).to.deep.equal({
+            valueType: 'date',
+            date: '2019-05-22T06:43:42Z' // this is correctly typed but will be an error at runtime
+        });
+    });
+    test('a complex record should be turned in the expected underlying representation', () => {
```

Comment:
@bame-da @meet-da You are probably interested in having a look at this test to understand how the helpers would work in practice. Line 142-159 are the version with the helpers, lines 161-211 are the current status, with the two being completely interchangeable at any level of granularity.

Reply:
Great improvement, thank you. From a user perspective, I'm left wondering why I have to annotate some values with `daml.x()` when that could be derived quite easily. I'd consider implicitly applying

- `daml.text` to strings
- `daml.record` to objects
- `daml.lst` to arrays

Reply:
This can be an improvement but before applying implicit conversions I'd like to wait. It saves only a few characters and I'm not entirely convinced of the pitfalls we may fall in or that we may let the user to fall in for a relatively small gain.

Reply:
> that could be derived quite easily.

This could be true on service calls, so maybe that is the place where we could intervene.

Otherwise, without any context, I have no way to derive that:
```
{
    a: [ 'Alice', 'Bob' ]
}
```
is a record with a list of DAML text values.